packages/@lwc/integration-karma/test/light-dom/lifecycle/index.spec.js

```diff
@@ -34,149 +65,192 @@ describe('standard slotting', () => {
+                ]);
+            });
         });
 
-        it('invokes lifecycle methods in correct order - dynamic', async () => {
-            const elm = createElement('x-dynamic-slot-forwarding', { is: DynamicSlotForwarding });
-            document.body.appendChild(elm);
-            await Promise.resolve();
-
-            // Initial connection
-            expect(window.timingBuffer).toEqual([
-                '0:connectedCallback',
-                '1:connectedCallback',
-                '2:connectedCallback',
-            ]);
-
-            resetTimingBuffer();
-
-            // Trigger vdom diffing
-            elm.showTop = true;
-            await Promise.resolve();
-
-            expect(window.timingBuffer).toEqual([
-                '3:connectedCallback',
-                '4:connectedCallback',
-                '5:connectedCallback',
-                '2:disconnectedCallback',
-                '1:disconnectedCallback',
-                '0:disconnectedCallback',
-            ]);
-
-            resetTimingBuffer();
-
-            // Trigger vdom diffing from top level
-            elm.topTop = 'bottom';
-            elm.topBottom = 'top';
-            await Promise.resolve();
-
-            expect(window.timingBuffer).toEqual([
-                '6:connectedCallback',
-                '5:disconnectedCallback',
-                '7:connectedCallback',
-                '3:disconnectedCallback',
-            ]);
-
-            resetTimingBuffer();
-
-            const { topSlot } = extractDataIds(elm);
-            // Trigger vdom diffing from forwarded slot level
-            topSlot.top = 'top';
-            topSlot.bottom = 'bottom';
-            await Promise.resolve();
-
-            expect(window.timingBuffer).toEqual([
-                '8:connectedCallback',
-                '6:disconnectedCallback',
-                '9:connectedCallback',
-                '7:disconnectedCallback',
-            ]);
-
-            resetTimingBuffer();
-
-            // vdom diffing after changing vnodes
-            elm.showTop = false;
-            await Promise.resolve();
-
-            // Synthetic custom element lifecycle does not fire disconnectedCallback correctly
-            expect(window.timingBuffer).toEqual(
-                !lwcRuntimeFlags.DISABLE_NATIVE_CUSTOM_ELEMENT_LIFECYCLE
-                    ? [
-                          '10:connectedCallback',
-                          '11:connectedCallback',
-                          '12:connectedCallback',
-                          '4:disconnectedCallback',
-                          '9:disconnectedCallback',
-                          '8:disconnectedCallback',
-                      ]
-                    : [
-                          '10:connectedCallback',
-                          '11:connectedCallback',
-                          '12:connectedCallback',
-                          '4:disconnectedCallback',
-                      ]
-            );
```

Comment:
This is actually a bugfix – we are now correctly firing the expected `disconnectedCallback`s in synthetic lifecycle mode.